packages/x-components/src/x-modules/experience-controls/store/actions/fetch-and-save-controls.action.ts

```diff
@@ -0,0 +1,29 @@
+import { createFetchAndSaveActions } from '../../../../store/utils/fetch-and-save-action.utils';
+import { ExperienceControlsActionContext } from '../types';
+
+const { fetchAndSave, cancelPrevious } = createFetchAndSaveActions<
+  ExperienceControlsActionContext,
+  any,
+  any
+>({
+  fetch({ dispatch }, request) {
+    return dispatch('fetchControls', request);
+  },
+  onSuccess({ commit }, controls) {
+    commit('setControls', controls);
```

Comment:
The success here should be returning an object containing the controls and the events, so you'll need another `commit` to set the events.